src/mpl2/README.md

```diff
@@ -1,20 +1,119 @@
+    [-num_bundled_ios  num_bundled_ios]
+    [-large_net_threshold  large_net_threshold]
+    [-signature_net_threshold  signature_net_threshold]
+    [-halo_width   halo_width] 
+    [-fence_lx fence_lx] 
+    [-fence_ly fence_ly]
+    [-fence_ux fence_ux]
+    [-fence_uy fence_uy]
+    [-area_weight  area_weight] 
+    [-outline_weight   outline_weight] 
+    [-wirelength_weight    wirelength_weight]
+    [-guidance_weight  guidance_weight]
+    [-fence_weight fence_weight] 
+    [-boundary_weight  boundary_weight]
+    [-notch_weight notch_weight]
+    [-macro_blockage_weight    macro_blockage_weight]
+    [-target_util  target_util]
+    [-target_dead_space    target_dead_space]
+    [-min_ar   min_ar]
+    [-snap_layer   snap_layer]
+    [-bus_planning_flag    bus_planning_flag]
+    [-report_directory report_directory]
+```
+
+#### Generic Parameters
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-max_num_macro`, `-min_num_macro` | Maximum/minimum number of macros in a cluster. (default 0, 0, integer). |
+| `-max_num_inst`, `-min_num_inst` | Maximum/minimum number of standard cells in a cluster. (default 0, 0, integer). |
+| `-tolerance` | A factor of $(1 + tol)$ is multiplied to the max, $(1 - tol)$ to the min number of macros/std cells in a cluster respectively. This is to improve robustness of hierarchical clustering. (default 0.1, float). |
+| `-max_num_level` | Maximum depth of physical hierarchical tree. (default 2). |
+| `-coarsening_ratio` | The larger the coarsening_ratio, the faster the convergence process. (default 10.0 ,float). | 
+| `-num_bundled_ios` | Specifies number of bundled pins for left, right, top, bottom boundary. (default 3, float). |
+| `-large_net_threshold` | Ignore nets with many connections during clustering, such as global nets. (default 50, integer). |
+| `-signature_net_threshold` | Minimum number of connections between two clusters for them to be identified as connected. (default 50, integer). |
+| `-halo_width` | Horizontal/vertical halo around macros (microns). (default 0.0, float). |
+| `-fence_lx`, `-fence_ly`, `-fence_ux`, `-fence_uy` | Defines the global fence bounding box coordinates. (defaults to core area coordinates). |
+| `-target_util` | Specifies the target utilisation of MixedCluster, and has higher priority than target_dead_space (default 0.25, float). |
+| `-target_dead_space` | Specifies the target deadspace percentage, which influences the utilisation of StandardCellCluster(default 0.05, float). |
+| `-min_ar` | Specifies the minimum aspect ratio $a$, or the ratio of its width to height of a StandardCellCluster from $[a, \frac{1}{a}]$ (default 0.33, float). |
+| `-snap_layer` | Snap macro origins to this routing layer track. (default 4, integer). | 
+| `-bus_planning_flag` | Flag to enable bus planning, recommendation is to turn on bus planning for SKY130, off for NanGate45/ASAP7.  (default false, bool). |
+| `-report_directory` | Save reports to this directory. |
```

Comment:
```suggestion
| `-max_num_macro`, `-min_num_macro` | Maximum/minimum number of macros in a cluster. The default value is `0` for both, and the allowed values are integers `[0, MAX_INT]`. |
| `-max_num_inst`, `-min_num_inst` | Maximum/minimum number of standard cells in a cluster. The default value is `0` for both, and the allowed values are integers `[0, MAX_INT]`. |
| `-tolerance` | A factor of $(1 + tol)$ is multiplied to the max, $(1 - tol)$ to the min number of macros/std cells in a cluster, respectively. This is to improve the robustness of hierarchical clustering. The allowed values are floats `[0, 1)`, and the default value is `0.1`. |
| `-max_num_level` | Maximum depth of physical hierarchical tree. The default value is `2`, and the allowed values are integers `[0, MAX_INT]`. |
| `-coarsening_ratio` | The larger the coarsening_ratio, the faster the convergence process. The allowed values are floats, and the default value is `10.0`. |
| `-num_bundled_ios` | Specifies the number of bundled pins for the left, right, top, and bottom boundaries. The default value is `3`, and the allowed values are integers `[0, MAX_INT]`. |
| `-large_net_threshold` | Ignore nets with many connections during clustering, such as global nets. The default value is `50`, and the allowed values are integers `[0, MAX_INT]`. |
| `-signature_net_threshold` | Minimum number of connections between two clusters to be identified as connected. The default value is `50`, and the allowed values are integers `[0, MAX_INT]`. |
| `-halo_width` | Horizontal/vertical halo around macros (microns). The allowed values are floats, and the default value is `0.0`. |
| `-fence_lx`, `-fence_ly`, `-fence_ux`, `-fence_uy` | Defines the global fence bounding box coordinates. The default values are the core area coordinates). |
| `-target_util` | Specifies the target utilization of `MixedCluster` and has higher priority than target_dead_space. The allowed values are floats, and the default value is `0.25`. |
| `-target_dead_space` | Specifies the target dead space percentage, which influences the utilization of `StandardCellCluster`. The allowed values are floats, and the default value is `0.05`. |
| `-min_ar` | Specifies the minimum aspect ratio $a$, or the ratio of its width to height of a `StandardCellCluster` from $[a, \frac{1}{a}]$. The allowed values are floats, and the default value is `0.33`. |
| `-snap_layer` | Snap macro origins to this routing layer track. The default value is 4, and the allowed values are integers `[1, MAX_LAYER]`). |
| `-bus_planning_flag` | Flag to enable bus planning. The recommendation is to turn on bus planning for SKY130 and off for NanGate45/ASAP7. The default value is disabled. |
| `-report_directory` | Save reports to this directory. |
```

Reply:
| `-tolerance` | A factor of $(1 + tol)$ is multiplied to the max, $(1 - tol)$ to the min number of macros/std cells in a cluster, respectively. This is to improve the robustness of hierarchical clustering. The allowed values are floats `[0, 1)`, and the default value is `0.1`. |

It essentially means that the min/max number of macros/std cells in a cluster is a soft threshold. For example the min can go even lower than the preset min.

Reply:
Maybe the following?
> Add a margin to the minimum and maximum number of macros/std cells in a cluster. For min, we multiply by (1 - `tol`), and for the max (1 + `tol`). This is to improve the robustness of hierarchical clustering. The allowed values are floats `[0, 1)`, and the default value is `0.1.`